web/core/components/workspace/sidebar/favorites/favorite-items/common/helper.tsx

```diff
@@ -0,0 +1,51 @@
+"use client";
+// lucide
+import { Briefcase, FileText, Layers } from "lucide-react";
+// types
+import { IFavorite, TLogoProps } from "@plane/types";
+// ui
+import { ContrastIcon, DiceIcon, FavoriteFolderIcon } from "@plane/ui";
+import { Logo } from "@/components/common";
+
+const iconClassName = `flex-shrink-0 size-4 stroke-[1.5] m-auto`;
+
+export const FAVORITE_ITEM_ICON: Record<string, JSX.Element> = {
+  page: <FileText className={iconClassName} />,
+  project: <Briefcase className={iconClassName} />,
+  view: <Layers className={iconClassName} />,
+  module: <DiceIcon className={iconClassName} />,
+  cycle: <ContrastIcon className={iconClassName} />,
+  folder: <FavoriteFolderIcon className={iconClassName} />,
+};
+
+export const getFavoriteItemIcon = (type: string, logo?: TLogoProps | undefined) => (
+  <>
+    <div className="hidden group-hover:flex items-center justify-center size-5">
+      {FAVORITE_ITEM_ICON[type] || <FileText />}
+    </div>
+    <div className="flex items-center justify-center size-5 group-hover:hidden">
+      {logo?.in_use ? (
+        <Logo logo={logo} size={16} type={type === "project" ? "material" : "lucide"} />
+      ) : (
+        FAVORITE_ITEM_ICON[type] || <FileText />
+      )}
+    </div>
+  </>
+);
+
+export const generateFavoriteItemLink = (workspaceSlug: string, favorite: IFavorite) => {
+  switch (favorite.entity_type) {
+    case "project":
+      return `/${workspaceSlug}/projects/${favorite.project_id}/issues`;
```

Comment:
nit: It would be cleaner to construct the project path
`/${workspaceSlug}/projects/${favorite.project_id}`